src/nvim/menu.c

```diff
@@ -1294,34 +1294,34 @@ static char_u *popup_mode_name(char *name, int idx)
 ///             allocated.
 ///
 /// @return a pointer to allocated memory.
-static char_u *menu_text(const char_u *str, int *mnemonic, char_u **actext)
+static char *menu_text(const char *str, int *mnemonic, char **actext)
   FUNC_ATTR_NONNULL_RET FUNC_ATTR_WARN_UNUSED_RESULT
   FUNC_ATTR_NONNULL_ARG(1)
 {
-  char_u *p;
-  char_u *text;
+  char *p;
+  char *text;
 
   // Locate accelerator text, after the first TAB
-  p = vim_strchr(str, TAB);
+  p = (char *)vim_strchr((char_u *)str, TAB);
   if (p != NULL) {
     if (actext != NULL) {
-      *actext = vim_strsave(p + 1);
+      *actext = (char *)vim_strsave((char_u *)p + 1);
     }
     assert(p >= str);
-    text = vim_strnsave(str, (size_t)(p - str));
+    text = (char *)vim_strnsave((char_u *)str, (size_t)(p - str));
```

Comment:
I think this is equivalent to `xstrndup` as well. we should make sure this, because ` (char *)vim_strnsave((char_u *)...)` linenoise will make no-one happy.

Reply:
Yeah :/ . I'll try to minimize increasing the number of casts as much as possible. Let me know if we should go with another strategy instead. I'm open to ideas :)

Interestingly enough, while `vim_strsave` essentially acts like a wrapper for `xstrdup`, `vim_strnsave` returns `(char_u *)strncpy(xmallocz(len), (char *)string, len)` with the following comment:

>  // strncpy is intentional: some parts of Vim use `string` shorter than `len`
   // and expect the remainder to be zeroed out.

So I assume `vim_strnsave` can be replaced with `xstrndup`, *except* for these "parts", which is a bit vague.

Reply:
We can otherwise slap a

```c
#define vim_strnsave(a,b) (char*)vim_strnsave((char_u*)a,b)
```

at the top of the file for now and remove it when the time is right?

Reply:
fixed

Reply:
so they are not the same, as `vim_strnsave` guarantees that `len+1` bytes are allocated even if the valid data is shorter which `xstrndup` doesn't. ideally the latter should work but to be safe rather than sorry we might want a variant that preserves `vim_strnsave` behavior, so we can move forward with the delet char_u project without worry.